src/backend/marsha/core/api/video.py

```diff
@@ -388,11 +393,77 @@ def initiate_upload(self, request, pk=None):
         # Reset the upload state of the video (don't use get_object()
         # as it does not lock the row)
         Video.objects.filter(pk=pk).update(
-            upload_state=defaults.PENDING, transcode_pipeline=defaults.AWS_PIPELINE
+            upload_state=defaults.PENDING, transcode_pipeline=defaults.PEERTUBE_PIPELINE
         )
 
         return Response(response)
 
+    @action(methods=["post"], detail=True, url_path="upload-ended")
+    # pylint: disable=unused-argument
+    def upload_ended(self, request, pk=None):
+        """Notify the API that the video upload has ended.
+
+        Calling the endpoint will start the transcoding process of the video.
+        The request should have a file_key in the body, which is the key of the
+        file uploaded on AWS. It will be used in the transcoding process to name
+        the generated files, and set the uploaded_on property.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+        # Ensure object exists and user has access to it
+        video = self.get_object()
+
+        serializer = serializers.UploadEndedSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        file_key = serializer.validated_data["file_key"]
+        # Should have the "tmp/{video_pk}/video/{stamp}" format
+        [tmp_dir, video_pk, video_dir, stamp] = file_key.split("/")
+
+        # avoid file key being forged
+        if pk != video_pk or tmp_dir != "tmp" or video_dir != "video":
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
`serializer.errors` should contain nothing here I guess. But maybe you can validate it at the serializer level adding a `validate_file_key` method as explain in the DRF documentation: https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation
You can inject the request in the serializer context to retrieve the `pk` in it.